src/components/draw/DrawDirective.js

```diff
@@ -161,165 +317,228 @@ goog.require('ga_map_service');
-              style: scope.options.drawStyleFunction
-            });
 
-            unDrawEnd = draw.on('drawend', function(evt) {
+              // Unregister the change event
+              ol.Observable.unByKey(deregFeatureChange);
+
               // Set the definitve style of the feature
-              var styles = scope.options.styleFunction(evt.feature);
-              evt.feature.setStyle(styles);
+              layer.getSource().addFeature(featureToAdd);
+              var styles = tool.style(featureToAdd);
+              featureToAdd.setStyle(styles);
               scope.$apply();
+              deactivateTool(lastActiveTool);
+              select.getFeatures().push(featureToAdd);
+              // Add final measure tooltips
+              if (tool.showMeasure) {
+                gaMeasure.addOverlays(map, layer, featureToAdd);
+                map.removeOverlay(distTooltip);
+                map.removeOverlay(areaTooltip);
+             }
             });
-
             map.addInteraction(draw);
           };
-
           var deactivateDrawInteraction = function() {
-            // Remove events
-            if (unDrawEnd) {
-              ol.Observable.unByKey(unDrawEnd);
-              unDrawEnd = undefined;
+            ol.Observable.unByKey(deregPointerMove2);
+            ol.Observable.unByKey(deregDrawStart);
+            ol.Observable.unByKey(deregDrawEnd);
+            map.removeOverlay(helpTooltip);
+            map.removeOverlay(distTooltip);
+            map.removeOverlay(areaTooltip);
+            if (draw) {
+              map.removeInteraction(draw);
             }
-            map.removeInteraction(draw);
-            draw = undefined;
-          };
-
-          // Set the select interaction
-          var activateSelectInteraction = function() {
-            deactivateDrawInteraction();
-            deactivateSelectInteraction();
-            deactivateModifyInteraction();
-            select.setActive(true);
-          };
-
-          var deactivateSelectInteraction = function() {
-            // Clearing the features updates scope.useXXX properties
-            select.getFeatures().clear();
-            select.setActive(false);
           };
 
-          // Set the modifiy interaction
-          var activateModifyInteraction = function() {
-            activateSelectInteraction();
-
-            modify.setActive(true);
-            if (!gaBrowserSniffer.mobile) {
-              viewport.on('mousemove', updateCursorStyleDebounced);
+          // Watchers
+          scope.$watch('isActive', function(active) {
+            if (active) {
+              activate();
+            } else {
+              deactivate();
             }
-          };
-
-          var deactivateModifyInteraction = function() {
-            modify.setActive(false);
-            viewport.unbind('mousemove', updateCursorStyleDebounced);
-          };
-
+          });
+          scope.$watch('popupToggle', function(active) {
```

Comment:
probably a relic of old times when we could select multiple features. Now the popup is closed when we deselect a feature so it seems useless now